test/integration/lib/update-lock.spec.ts

```diff
@@ -293,6 +296,27 @@ describe('lib/update-lock', () => {
 				'using lockOverride gave lock ownership to the callback, so they should now be deleted',
 			);
 		});
+
+		it('should not try to take Supervisor locks that are already taken', async () => {
+			// Take Supervisor locks
+			await takeLocksWithUid(updateLock.LOCKFILE_UID);
+
+			// Attempt to take locks again from another context,
+			// which should not throw an error as locks that are already
+			// taken should not be taken again by the Supervisor.
+			await updateLock
+				.lock(testAppId, { force: false }, () => Promise.resolve())
+				.catch(async (err) => {
+					expect(
+						await updateLock.getLocksTaken(),
+						'Locks taken in initially should still be held',
+					).to.have.length(1);
+					expect.fail(`update lock should not throw error but got: ${err}`);
+				});
```

Comment:
I think would be enough

```suggestion
			await expect(updateLock
				.lock(testAppId, { force: false }, () => Promise.resolve())).to.not.be.rejected;
```